File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        int[] results = new int[nums.Length];
        int num = 1;
        
        for(int i=0; i<nums.Length;i++){
            results[i] = num;
            num *= nums[i];    
        }
        num = 1;
        for(int i=nums.Length-1; i>=0; i--){
            results[i] *= num;
            num *= nums[i];
        }
        return results;
    }
}

File: Data Structures & Algorithms/sliding-window-maximum/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    maxSlidingWindow(nums, k) {
        if(nums.length === 0 || k === 0){
            return [];
        }
        const result = [];
        for(let i = 0; i<=nums.length-k; i++){
            const currentWindow = nums.slice(i, i+k);
            result.push(Math.max(...currentWindow));
        }
        return result;
    }
}
